src/main/java/org/vaadin/viritin/ListContainer.java

```diff
@@ -482,15 +485,16 @@ private PropertyComparator(Object[] propertyId, boolean[] ascending) {
         public int compare(T o1, T o2) {
             for (int i = 0; i < propertyId.length; i++) {
                 String currentProperty = propertyId[i].toString();
-                Comparator<T> currentComparator
-                        = new BeanComparator<T>(currentProperty,
-                                getUnderlyingComparator(currentProperty));
+                Comparator underlyingComparator = getUnderlyingComparator(currentProperty);
+                Comparator currentComparator = underlyingComparator != null ? underlyingComparator : ComparableComparator.getInstance();
 
                 if (!ascending[i]) {
-                    currentComparator = new ReverseComparator(currentComparator);
+                    currentComparator = currentComparator.reversed();
```

Comment:
reversed is default method from JDK8? Can you change this back in to using ReverseComparator as Viritin still supports JDK7.